store/src/store.rs

```diff
@@ -168,21 +168,29 @@ impl<DB: Database> KVStore<DB> {
         }
     }
 
-    pub fn range<R>(&self, range: R) -> Range<'_, R, DB>
+    pub fn range<'a, R>(&'a self, range: R) -> impl Iterator<Item = (Vec<u8>, Vec<u8>)> + 'a
     where
-        R: RangeBounds<Vec<u8>> + Clone,
+        R: RangeBounds<Vec<u8>> + Clone + 'a,
     {
-        //TODO: this doesn't iterate over cached values
-        // let tx_cached_values = self.tx_cache.range(range.clone());
-        // let block_cached_values = self.block_cache.range(range.clone());
-        // let persisted_values = self.persistent_store.range(range.clone());
-
-        // MergedRange::merge(
-        //     tx_cached_values,
-        //     MergedRange::merge(block_cached_values, persisted_values),
-        // );
-
-        self.persistent_store.range(range)
+        let tx_cached_values = self.tx_cache.range(range.clone());
+        let block_cached_values = self.block_cache.range(range.clone());
+        let mut persisted_values = self
+            .persistent_store
+            .range(range.clone())
+            .collect::<BTreeMap<_, _>>();
```

Comment:
This will pull all values from the persistent store (within the range) into memory. We want to avoid doing this.